METHOD_NAMING_PROPOSAL.md

```diff
@@ -0,0 +1,412 @@
+# v5 Method Naming Proposal
+
+## Overview
+
+In v5 of the Braintree Android SDK, we are introducing a new integration pattern that includes a 
```

Comment:
Can we get more info on why the move to this `Launcher` class is being proposed? What are the potential benefits/setbacks to this design change?

Reply:
Yeah @hollabaq86 this has roots in Android's [Activity Result API](https://developer.android.com/training/basics/intents/result). This pattern allows us to further decouple feature clients from Android, which is a big win for merchants because they don't have unnecessary restrictions imposed on them.

We introduced [Client Token Provider](https://github.com/braintree/braintree_android/blob/main/v4.9.0%2B_MIGRATION_GUIDE.md#client-token-provider) in `v4` to provide a BraintreeFragment-like `v3` interface. There were benefits to providing a single entry point into each feature via Feature Clients before `startActivityForResult()` was deprecated. With `v4`, we decided to fully encapsulate the new Activity Result API within each feature client. This pattern works well for 90% of merchant integrations, but it can get complex when merchants integrate with the Braintree SDK in a headless manner i.e. without a UI.

All in all, complexity is conserved because we're adding a `Launcher` concept and (eventually) removing the `ClientTokenProvider` concept. I'd be down to deep dive into this some time! This pattern is the result of listening to our merchants and it will make onboarding `v5` a lot smoother.